pyvrp/cpp/ProblemData.h

```diff
@@ -162,6 +185,62 @@ class ProblemData
         ~Client();
     };
 
+    /**
+     * ClientGroup(clients: list[int], required: bool = True)
+     *
+     * A client group that imposes additional restrictions on visits to clients
+     * in the group.
+     *
+     * .. note::
+     *
+     *    Only mutually exclusive client groups are supported for now.
+     *
+     * Parameters
+     * ----------
+     * clients
+     *     The clients in the group.
+     * required
+     *     Whether visiting this client group is required.
+     *
+     * Attributes
+     * ----------
+     * clients
+     *     The clients in the group.
+     * required
+     *     Whether visiting this client group is required.
+     * mutually_exclusive
+     *     When ``True``, exactly one of the clients in this group must be
+     *     visited if the group is required, and at most one if the group is
+     *     not required.
+     */
+    class ClientGroup
+    {
+        std::vector<size_t> clients_;  // clients in this group
```

Comment:
Some questions about the `clients_` variable:
- Does it matter for the implementation if it's ordered?
- Does it matter if there are client duplicates?
- Would we rather use a `set` instead of a vector here?

Reply:
- No.
- A little cause we iterate over these and if they're bigger than they need to be we do extra work for no gain.
- Iterating a set is (much) slower than iterating a vector. We spend most time iterating, not checking membership. I think we can add a check to `addClient` that throws when the client is already in the group.